code/obj/cryotron.dm

```diff
@@ -361,9 +363,58 @@
 		if (!exit_prompt(user))
 			return ..()
 
+	/// Handling dragging players in to cryo, mainly for silicon players.
 	MouseDrop_T(atom/target, mob/user as mob)
-		if (ishuman(target) && isrobot(user) && BOUNDS_DIST(src, user) == 0 && BOUNDS_DIST(src, target) == 0 && BOUNDS_DIST(user, target) == 0)
-			insert_prompt(target, user)
+		if (!ishuman(target) && !isrobot(target))
+			return
+
+		if (BOUNDS_DIST(src, user) != 0)
+			return
+
+		if (BOUNDS_DIST(src, target) != 0)
 			return
+
+		if (BOUNDS_DIST(user, target) != 0)
+			return
+
+		insert_prompt(target, user)
 		return ..()
 
+	/// Override for handling all interactions with the cryo chamber
+	Click()
+		if (!usr)
+			return
+
+		if (isAIeye(usr))
+			boutput(usr, "<span class='alert'>An incorporeal manifestation of an artificial intelligence's presence can't enter \the [src]!</span>")
+			return
+
+		if (!can_act(usr))
+			return
+
+		if (!in_interact_range(src, usr))
+			return
+
+		if (isdead(usr))
+			return
+
+		if (isobserver(usr))
+			return
+
+		if (isintangible(usr))
+			return
+
+		if (issilicon(usr))
+			enter_prompt(usr)
```

Comment:
instead of using multiple ifs chained together, you could probably combine them into one if statement.
`if (this || that || thisthing || otherthing)`

Reply:
Grouped up some of the contracts